src/svg/path/path.py

```diff
@@ -49,6 +49,31 @@ def __ne__(self, other):
             return NotImplemented
         return not self == other
 
+    def __add__(self, other):
+        self.start += other
+        self.end += other
+
+    def __sub__(self, other):
+        self.start -= other
+        self.end -= other
+
+    def __mul__(self, other):
```

Comment:
Although multiplications on complex numbers are cool, they don't really translate into anything particularly useful in graphics. I don't think we need it.

Reply:
Multiplication does scaling. If you take that element and you multiply by 2 it scales by 2.

Reply:
It also moves it by two, so that's not what you want in most cases.

I think we probably should implement this by first implementing the matrix transformation, and then possibly implementing others as shortcuts.

Reply:
It doesn't move it by 2. It multiplies the .imag and .real by that value which is a uniform scaling operation, relative to the origin. To scaled based on a given point you'd subtract by a complex, multiply by a real, then add the original complex again. 

(x + yj) * s = (s*x, s*y). That's a properly scaled point.

Reply:
Yes, and that's what is necessary. If you have a path that makes a square on point 1+1j, 1+2j, 2+2j and 2+1j and you multiply that by 2 you get the points 2+2j, 2+4j, 4+4j and 4+2j.

That's scaled AND moved, and that's not a useful graphical operation.

Reply:
@regebro, nope. That's scaled.

You are confusing scaled with scaled relative to a location. Whereas the operation scaled is relative to the origin. To perform a scale relative to the center you would do path 
path -= 1.5 + 1.5j
path *= 2
path += 1.5 + 1.5j

That is scale and scale is by definition a *very* useful graphical operation. I think the confusion is in thinking scale cannot move elements. It totally can. I'm not sure how you would make all point relative to other points larger without moving the points around.

A rectangle at (1,1), (1,2), (2,2), (2,1)  scaled by 2 has points (2,2), (2,4), (4,4), (4,2) that's not an argument for wrongness, that's the definition of scaled in a geometric sense. All scale operations are relative to the origin. You need to do "transform(-x,-y), scale(s), transform(x,y)" to do this relative to a point but that's really just move the origin, scale, move the origin back.

Reply:
If you want to wrap your head around it, consider the distance between any point and any other point in the original versus scaled version. You will find they are in *every* case exactly twice as far away.

> from svg.elements import *
> print(Path("M1,1 1,2 2,2 2,1z") * "scale(2)")

`M 2,2 L 2,4 L 4,4 L 4,2 Z`

Reply:
I have wrapped my head around it. Thanks.